subprojects/publish/src/main/java/org/gradle/api/publish/internal/validation/DuplicatePublicationTracker.java

```diff
@@ -48,7 +48,14 @@ public synchronized void checkCanPublish(Project project, PublicationInternal<?>
         ModuleVersionIdentifier projectIdentity = publication.getCoordinates();
         for (PublicationWithProject previousPublicationWithProject : published.get(repositoryKey)) {
             if (previousPublicationWithProject.getPublication().getCoordinates().equals(projectIdentity)) {
-                LOG.warn("Multiple publications with coordinates '" + publication.getCoordinates() + "' are published to repository '" + repositoryName + "'. The publications " + previousPublicationWithProject + " and " + publicationWithProject + " will overwrite each other!");
+                String firstPublication = previousPublicationWithProject.toString();
```

Comment:
An alternative would be to add the two Strings to a list and then sort the list. Would that be simpler?

Reply:
As there are only two values, that seems overkill IMO